MoreLinq/StartsWith.cs

```diff
@@ -0,0 +1,70 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Andreas Gullberg Larsen. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///     Determines whether the beginning of <paramref name="first" /> matches <paramref name="second" />.
+        /// </summary>
+        /// <typeparam name="T">Type of elements.</typeparam>
+        /// <param name="first">The sequence to check.</param>
+        /// <param name="second">The sequence to compare to.</param>
+        /// <returns>
+        ///     <c>true</c> if <paramref name="first" /> begins with elements equivalent to <paramref name="second" />.
+        /// </returns>
+        /// <remarks>
+        ///     This is the <see cref="IEnumerable{T}" /> equivalent of <see cref="string.StartsWith(string)" /> and
+        ///     it calls <see cref="IEqualityComparer{T}.Equals(T,T)" /> using <see cref="EqualityComparer{T}.Default"/> on pairs of elements at the same index.
+        /// </remarks>
+        public static bool StartsWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        {
+            return StartsWith(first, second, EqualityComparer<T>.Default);
+        }
+
+        /// <summary>
+        ///     Determines whether the beginning of <paramref name="first" /> matches <paramref name="second" />.
```

Comment:
Method summaries should be descriptive and avoid referring to any parameters because the rest of the documentation details may not be available for context, e.g., in a summary page like [table of content of methods](https://morelinq.github.io/2.0/ref/api/html/T_MoreLinq_MoreEnumerable.htm). Some like the following would be better but feel free to re-word.

> Determines whether the starting elements of the first sequence match those of a second sequence.

Also, the two overloads have *identical* summaries. Please [see docs of other MoreLINQ methods](https://morelinq.github.io/2.0/ref/api/html/T_MoreLinq_MoreEnumerable.htm) and [LINQ](https://msdn.microsoft.com/en-us/library/system.linq.enumerable(v=vs.100).aspx) itself for inspiration.

Ditto for `EndsWith`.